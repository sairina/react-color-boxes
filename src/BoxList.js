import React, { useState } from 'react';
import NewBoxForm from './NewBoxForm';
import Box from './Box';
import uuid from 'uuid/v4';

function BoxList() {
  //state
  const [boxes, setBoxes] = useState([]);

  //render boxes array
  //will need to iterate over boxes
  const renderBoxes = () => {
    return (
      <div className="BoxList-renderBoxes">
        {boxes.map(box => (
          <Box key={box.id}
            backgroundColor={box.backgroundColor}
            width={box.width}
            height={box.height}
          />
        ))}
      </div>
    );
  };

  //addBox to array of boxes
  const addBox = box => {
    let newBox = { ...box, id: uuid() };
    setBoxes(boxes => [...boxes, newBox]);
  };
  
  return (
    <div className="BoxList">
      <NewBoxForm addBox={addBox} />
      {renderBoxes()}
    </div>
  );
}

export default BoxList;