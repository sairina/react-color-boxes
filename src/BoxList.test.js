import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot - new BoxList", function() {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("it can add a new box", function() {
  const {getByLabelText, queryByText } = render(<BoxList />);

  expect(queryByText("X")).not.toBeInTheDocument();

  const backgroundColorInput = getByLabelText("Background Color:");
  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const submitButton = queryByText("Add new box");

  fireEvent.change(backgroundColorInput, { target: { value: "blue"}});
  fireEvent.change(widthInput, { target: { value: 50}});
  fireEvent.change(heightInput, { target: { value: 50}});
  fireEvent.click(submitButton);

  expect(queryByText("X")).toBeInTheDocument();
});

it("it can delete a box", function() {
  const {getByLabelText, queryByText } = render(<BoxList />);

  const backgroundColorInput = getByLabelText("Background Color:");
  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const submitButton = queryByText("Add new box");

  fireEvent.change(backgroundColorInput, { target: { value: "blue"}});
  fireEvent.change(widthInput, { target: { value: 50}});
  fireEvent.change(heightInput, { target: { value: 50}});
  fireEvent.click(submitButton);

  expect(queryByText("X")).toBeInTheDocument();

  const removeButton = queryByText("X");

  fireEvent.click(removeButton);

  expect(queryByText("X")).not.toBeInTheDocument();
});
