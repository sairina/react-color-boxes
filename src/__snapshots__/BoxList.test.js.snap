// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - new BoxList 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <div
      class="container"
    >
      <form
        class="NewBoxForm"
      >
        <label
          for="backgroundColor"
        >
          Background Color: 
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          value=""
        />
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          id="width"
          name="width"
          value=""
        />
        <label
          for="height"
        >
          Height: 
        </label>
        <input
          id="height"
          name="height"
          value=""
        />
        <button
          class="NewBoxForm-button"
        >
          Add new box
        </button>
      </form>
    </div>
    <div
      class="BoxList-renderBoxes"
    />
  </div>
</DocumentFragment>
`;
